ignite/metrics/ssim.py

```diff
@@ -157,6 +159,15 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
                 f"Expected y_pred and y to have BxCxHxW shape. Got y_pred: {y_pred.shape} and y: {y.shape}."
             )
 
+        if y_pred.dtype == torch.uint8 or y.dtype == torch.uint8:
+            if self.data_range != 255:
+                warnings.warn(
+                    "dtypes of the input tensors are torch.uint8 but data range is not set to 255.", RuntimeWarning
+                )
```

Comment:
I'm not a big fan of showing a warning in this case. I would follow the same strategy as skimage where they just transform to floating point. So, I would do the following:
```
if not y.is_floating_point():
    y = y.float()
if not y_pred.is_floating_point():
    y_pred = y_pred.float()
```

Reply:
Maybe we should put a default value to the data_range argument too? Most of the time we are dealing with [0; 1] images.

Reply:
I do not have a strong opinion on this. We have to figure out if there wont be any negative impact for the users in this case.

Reply:
I think the skiimg approach is quite good: default to data_range = 1.0, and warn if the detected type is of integer and that data range is still set to 1